packages/client/src/comp/Address/Modal.tsx

```diff
@@ -0,0 +1,146 @@
+  const [ streetNumber, setStreetNumber ] = React.useState<string | undefined>()
+
+  // Used to imperatively position <Select/> for election officials
+  const [ width, setWidth ] = React.useState<number | string>(0)
+  const [ top, setTop ] = React.useState<number | string>(0)
+  const [ left, setLeft ] = React.useState<number | string>(0)
+
+  const state = fields.state as ImplementedState
+
+  const fetchContacts = React.useCallback(async () => {
+    setFetchingData(true)
+    const resp = await client.fetchContacts(state)
+    if (resp.type === 'data') {
+      setContacts(resp.data)
+      setNewContact(resp.data[0])
+    }
+    setFetchingData(false)
+  }, [setContacts, setFetchingData, state])
+
+  React.useEffect(() => {
+    // Loads the initial data when the modal is open for the first time
+    if (isOpen && !contacts) {
+      fetchContacts()
+    }
+    // Assigns the default value for streetNumber if possible
+    if (isOpen && !streetNumber && streetNumbers) {
+      setStreetNumber(streetNumbers[0])
+    }
+  }, [streetNumbers, streetNumber, address, contacts, fetchContacts, isOpen])
+
+  const handleSubmit = async (e: React.MouseEvent<HTMLButtonElement, MouseEvent>) => {
+    e.persist()
+    const address = addressPartsToAddress(fields)
+
+    if (streetNumber) {
+      address.streetNumber = streetNumber
+    }
+
+    if (newContact) {
+      setFetchingData(true)
+      const resp = await client.getContact(state, newContact)
+      if (resp.type === 'data') {
+        setContact(resp.data)
+        address.county = resp.data.county
+        setAddress(address)
+      }
+      setFetchingData(false)
+    }
+
+    pushState(state)
+    setOpen(false)
+  }
+
+  // Due to CSS/HTML limitation we can't have styled Select/Options that
+  // displays their content outside their boundaries. We render StyledSelect
+  // on a separate div from the Modal and uses this ref to assign its position
+  // https://reactjs.org/docs/hooks-faq.html#how-can-i-measure-a-dom-node
+  const positionRef = React.useCallback((node: HTMLDivElement) => {
+    setTimeout(
+      () => {
+        if (node !== null) {
+          const { top, x, width} = node.getBoundingClientRect()
+          setTop(top)
+          setLeft(x)
+          setWidth(width)
+        }
+      },
+      // A bit after the animation duration to ensure getBoundingClientRect
+      // gets everything correctly
+      505,
+    )
+  }, [])
+
+  // Returns true when this modal needs to solve both the lack of street
+  // number and of election official
+  const bothErrors = !contact && !address?.streetNumber
+
+  return <>
+    <StyledModal isOpen={isOpen}>
+      <AfterModalAnimation>
+        <h3>Geocode API Failed</h3>
+        <p>Please confirm or change the information below.</p>
+
+        <div
+          ref={positionRef}
+          // Height is either the height of one or two <Select/>
+          style={{ width: '100%', height: bothErrors ? 134 : 67 }}
+        />
```

Comment:
Had to keep the title and message of this modal as brief as possible.

I tried my best to implement two distinct Select components using the fixed positioned trick to display their options on top of the modal, however in this modal things were a bit diffferent.

Since the content would scroll if we use bigger message bodies the `Select` elements would stay in the same place (not following the scroll). To solve this I had created a effect that used to track if the `StyledModal` had any scroll event, if so manually update the position of each `Select`--but this was costly, on mobile devices if we scrolled too fast we could see some jittery positioning (so I decided to drop this as well).

To avoid using the default style for `Select` in only a single part of our application i had to shorten the amount of characters in this error message, the result now works in all devices and we are still able to keep the styled `Select` element